.github/workflows/publish_pkgci_mi300.yml

```diff
@@ -0,0 +1,44 @@
+name: Publish manylinux_x86_64 image
```

Comment:
Update the job name to match the image name. The name will show up here, so each should be unique: https://github.com/iree-org/base-docker-images/actions

![image](https://github.com/user-attachments/assets/ff0783b3-4d3b-432d-8c44-77f6761b66f2)